src/Nerdbank.Streams/MultiplexingStream.Channel.cs

```diff
@@ -916,17 +994,28 @@ private async Task AutoCloseOnPipesClosureAsync()
 
             private void Fault(Exception exception)
             {
-                if (this.TraceSource?.Switch.ShouldTrace(TraceEventType.Critical) ?? false)
+                this.mxStreamIOReader?.CancelPendingRead();
+
+                // If the channel has already been disposed then only cancel the reader
+                lock (this.SyncObject)
                 {
-                    this.TraceSource!.TraceEvent(TraceEventType.Critical, (int)TraceEventId.FatalError, "Channel Closing self due to exception: {0}", exception);
+                    if (this.isDisposed)
+                    {
+                        return;
+                    }
                 }
 
+                // Set the faulting exception to the current exception if it isn't already set
                 lock (this.SyncObject)
                 {
                     this.faultingException ??= exception;
                 }
 
-                this.mxStreamIOReader?.CancelPendingRead();
+                if (this.TraceSource?.Switch.ShouldTrace(TraceEventType.Critical) ?? false)
+                {
+                    this.TraceSource!.TraceEvent(TraceEventType.Critical, (int)TraceEventId.FatalError, "Channel faulting self due to exception: {0}", exception.Message);
```

Comment:
Similar feedback to above: avoid `FatalError` at the channel level. And `TraceEventType.Critical` is probably over-selling it as well.

Reply:
Changed the code to 0 and set to the event type to a warning.